### DISCUSSION TITLE
How to make a super cape for all completion-at-point-functions

### DISCUSSION THREAD

**Discussion by kiennq** [minad/cape#118] [Q&A] [answered]
I'm looking for a similar thing like `(setq company-backends '(company-capf :with company-dabbrev-code company-yasnippet :separate))` in `company-mode`.
That will make the make the candidates from the current CAPF shows up together with the one from dabbrev and snippets, **regardless of how the `completion-at-point-functions`** has changed.

**Comment by minad** [comment, marked as answer]
You can execute something like this in a buffer to extend the existing Capfs with dabbrev/yasnippet:

~~~ elisp
(defvar-local orig-capfs nil)
(unless orig-capfs
  (setq-local orig-capfs completion-at-point-functions))
(setq-local completion-at-point-functions
            (mapcar
             (lambda (f) (cape-capf-super f :with cape-dabbrev yasnippet-capf))
             orig-capfs))
~~~
